src/worker.rs

```diff
@@ -64,29 +60,23 @@ fn worker_impl(
         .insert_source(rx_chan, |event, _, state| {
             if let channel::Event::Msg(event) = event {
                 match event {
-                    Command::StorePrimary(contents) => {
-                        state.store_selection(SelectionTarget::Primary, contents);
-                    },
-                    Command::Store(contents) => {
-                        state.store_selection(SelectionTarget::Clipboard, contents);
-                    },
-                    Command::Load if state.data_device_manager_state.is_some() => {
-                        if let Err(err) = state.load_selection(SelectionTarget::Clipboard) {
-                            let _ = state.reply_tx.send(Err(err));
-                        }
+                    Command::Exit => state.exit = true,
+                    Command::Store(data, target) => {
+                        state.store_selection(target, data);
                     },
-                    Command::LoadPrimary if state.data_device_manager_state.is_some() => {
-                        if let Err(err) = state.load_selection(SelectionTarget::Primary) {
+                    Command::Load(mime_types, target)
+                        if state.data_device_manager_state.is_some() =>
+                    {
+                        if let Err(err) = state.load_selection(target, &mime_types) {
```

Comment:
That doesn't follow the old code, because it doesn't check primary selection clipboard, which may be absent.

Reply:
I'm not sure what you mean. I don't think the old code checked primary_selection_manager_state but I can add a check for that.

Reply:
Hm, the code I wrote is fool-proof and actually doesn't try to unwrap anything. I'm not even sure whether the first check is needed for data device, given that it's also checked, kind of.

Though, maybe won't hurt.